### PR TITLE
🐛 Patch: Backport Provider to v1alpha6

### PR DESCRIPTION
**Pull request by Jeremy-Boyle**
**What this PR does / why we need it**:
This is needed for the PR to `release-7.0` that allows provider to be back ported to v1alpha6.

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**TODOs**:

- [x] squashed commits
- [ ] Merge Backport PR: https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1529
- if necessary:
  - [ ] includes documentation
  - [x] adds unit tests

/hold

### CODE REVIEW

**Comment by Jeremy-Boyle** [comment]
> [CLA Missing ID](https://api.easycla.lfx.linuxfoundation.org/v2/repository-provider/github/sign/20677499/137236252/1530/#/?version=2) [CLA Not Signed](https://api.easycla.lfx.linuxfoundation.org/v2/repository-provider/github/sign/20677499/137236252/1530/#/?version=2)
> 
> * ❌ The commit ([d8ce8ae](https://github.com/kubernetes-sigs/cluster-api-provider-openstack/commit/d8ce8ae5c1d4a58d0c29a38b7945ce8fc6611b63)). This user is missing the User's ID, preventing the EasyCLA check. [Consult GitHub Help](https://help.github.com/en/github/committing-changes-to-your-project/why-are-my-commits-linked-to-the-wrong-user) to resolve.For further assistance with EasyCLA, [please submit a support request ticket](https://jira.linuxfoundation.org/servicedesk/customer/portal/4).

Not sure why its complaining about this but not in my other PR.

File: api/v1alpha5/conversion.go [hunk 1]
```diff
@@ -237,6 +237,6 @@ func Convert_v1alpha5_OpenStackMachineSpec_To_v1alpha7_OpenStackMachineSpec(in *
 }
 
 func Convert_v1alpha7_APIServerLoadBalancer_To_v1alpha5_APIServerLoadBalancer(in *infrav1.APIServerLoadBalancer, out *APIServerLoadBalancer, s conversion.Scope) error {
-	// Provider have been added in v1alpha7 but have no equivalent in v1alpha5
+	// Provider was orignally added in v1alpha7, but was backported to v1alpha6, but has no equivalent in v1alpha5
```
Comment (mdbooth) [inline api/v1alpha5/conversion.go:240]: linter is complaining about this typo
```suggestion
	// Provider was originally added in v1alpha7, but was backported to v1alpha6, but has no equivalent in v1alpha5
```

Comment (mdbooth) [reply api/v1alpha5/conversion.go]: It let me add that commit! Not done that before. Lets see if it makes the tests pass. You'll want to squash it anyway.

**Comment by mdbooth** [comment]
/approve

Please can you squash my typo fix before removing the hold?

**Comment by dulek** [comment]
/lgtm

I think this one is an API change simple enough for me to review.

**Comment by mdbooth** [comment]
@Jeremy-Boyle You might want to remove the hold (with /hold cancel) and it'll merge automatically when the tests pass.

**Comment by Jeremy-Boyle** [comment]
> /test pull-cluster-api-provider-openstack-e2e-test
> 
> Hopefully just a flake 🤞 If it fails again we'll have to look properly, because we don't get too many flakes.

Looks like its still failing
/retest

**Pull request merged**